app/models/miq_report_result.rb

```diff
@@ -71,7 +73,7 @@ def html_rows(options = {})
     end
     update_attribute(:last_accessed_on, Time.now.utc)
     purge_for_user
-    html_details.apply_legacy_finder_options(options.merge(:order => "id asc")).collect(&:data)
+    html_details.order("id asc").offset(options[:offset]).limit(options[:limit]).collect(&:data)
```

Comment:
Doesn't this ignore any other options that were passed in?  Or do to the callers never do that?